### PR TITLE
Fix for #3374

### PR DESCRIPTION
**Pull request by dbankov-vmware**
Introduce -platformlib option to control libraries added automatically for Windows targets.

### CODE REVIEW

File: driver/linker-msvc.cpp [hunk 1]
```diff
@@ -85,6 +94,26 @@ void addSanitizerLibs(std::vector<std::string> &args) {
 
 //////////////////////////////////////////////////////////////////////////////
 
+static std::vector<std::string> getPlatformLibNames() {
```
Comment (kinke) [inline driver/linker-msvc.cpp:97]: Thx for following up. - Please move this to `linker.cpp` (perhaps as `getExplicitPlatformLibs()` returning an `llvm::Optional<std::vector<string>>`) as we'll support this option for Posix targets too (see [here](https://github.com/ldc-developers/ldc/blob/9eb9f5a568ec21f84449d039132e25ce2abfc238/driver/linker-gcc.cpp#L630-L681)). The tokenizing can then be extracted as little helper, used for both `-defaultlib` and `-platformlib`.

Comment (dbankov-vmware) [reply driver/linker-msvc.cpp]: Finally got some time to do this. I'm sorry for the delay.
Please take a look and let me know if I should change something.

File: tests/linking/platformlib.d [hunk 2]
```diff
@@ -0,0 +1,21 @@
+/* Make sure -platformlib overrides the default platform libraries list.
+ * We only care about the platform libs in the linker command line;
+ * make sure linking fails in all cases (no main()) as linking would
+ * fail without the platform libraries anyway. Finally this option is
+ * relevant only for windows targets so make sure we target Windows with
+ * -mtriple=x86_64-unknown-windows-coff.
+ */
+
+// RUN: not %ldc -v -mtriple=x86_64-unknown-windows-coff -platformlib= %s | FileCheck %s
```
Comment (JohanEngelen) [inline tests/linking/platformlib.d:9]: can you also add a testcase where you pass some strings through the new option?

File: driver/linker-msvc.cpp (hunk 1 above)
Comment (kinke) [reply driver/linker-msvc.cpp]: No worries. - I've pushed a commit, slightly revising a few things, extending it to Posix targets too as well as extending the tests as requested by Johan.

Comment (dbankov-vmware) [reply driver/linker-msvc.cpp]: Thank you very much Kinke!
I was going to take care of this over the weekend as I've been quite busy at work lately...

**Comment by kinke** [comment]
[I've apparently hit some GitHub bug while merging this - this shouldn't have been a squash-merge, plus I've customized the merge commit msg - then got an error and pressed a retry button, which squash-merged it without further confirmation...]

**Comment by dbankov-vmware** [comment]
I'm not particularly proud of my part of the change so IMO it is better that things happened this way :).

**Comment by kinke** [comment]
Well, if it wasn't for you and your initiative, we wouldn't have this. :) - This wasn't a squash merge, but a normal merge with the actual merge commit (title + msg) looking like a squash-merge-commit...

**Comment by denizzzka** [comment]
What exactly is the difference between `--platformlib` and `--defaultlib`?

**Comment by kinke** [comment]
IIRC, mainly that `-defaultlib` doesn't apply to `-betterC`, and that the default lib names are affected by `-link-defaultlib-{shared,debug}` (extra suffix). And for a simpler overriding of the default libs, e.g., `-defaultlib=druntime-ldc` when wanting to exclude Phobos, or `-defaultlib=phobos2-ldc-lto,druntime-ldc-to` to use the LTO variants.

**Pull request merged**